JavaScript/JavaScript.sublime-syntax

```diff
@@ -8,12 +8,16 @@ file_extensions:
 first_line_match: ^#!\s*/.*\b(node|js)\b
 scope: source.js
 variables:
-  identifier: '[_$[:alpha:]][_$[:alnum:]]*'
-  constant_identifier: '[[:upper:]][_$[:digit:][:upper:]]*\b'
-  dollar_only_identifier: '\$(?![_$[:alnum:]])'
-  dollar_identifier: '(\$)[_$[:alnum:]]+'
-  func_lookahead: '\s*\b(async\s+)?function\b'
-  arrow_func_lookahead: '\s*(\basync\s*)?([_$[:alpha:]][_$[:alnum:]]*|\(([^()]|\([^()]*\))*\))\s*=>'
+  identifier_start: '[_$\p{L}\p{Nl}]'
+  identifier_part: '[_$\p{L}\p{Nl}\p{Mn}\p{Mc}\p{Nd}\p{Pc}\x{200C}\x{200D}]'
+  identifier_break: (?!{{identifier_part}})
+
+  identifier: '{{identifier_start}}{{identifier_part}}*{{identifier_break}}'
+  constant_identifier: '[[:upper:]]{{identifier_part}}*{{identifier_break}}'
+  dollar_only_identifier: '\${{identifier_break}}'
+  dollar_identifier: '(\$){{identifier_part}}*{{identifier_break}}+'
```

Comment:
@Thom1729 This `dollar_identifier` regular expression seems to cause errors when the JavaScript language definition is used within [`syntect`](https://github.com/trishume/syntect/).

In particular, it uses `+` on `{{identifier_break}}`, which is a negative-lookahead group: `(?!...)+`. Is this valid regex syntax? Could/should the `+` be removed?

Reply:
It's valid but redundant. I don't know why syntect doesn't handle this, but if you open up a PR to remove the `+` I'll endorse it.

Reply:
(for anyone arriving here, @sharkdp raised a PR: https://github.com/sublimehq/Packages/pull/1525)